docs/table/operations.rst

```diff
@@ -1088,6 +1088,24 @@ problem of "fuzzy" matching of key column values, where instead of an exact matc
 we require only an approximate match. This is supported using the ``join_funcs``
 argument.
 
+.. warning::
+
+   The coordinate and distance table joins discussed in this section are most
+   applicable in the case where the relevant entries in at least one of the
+   tables are all separated from one another by more than join distance. If this
+   is not satisfied then the join results may be unexpected.
+
+   This is a consequence of the algorithm which assigns a common source
+   identifier to the left and right entries for every pair which is within the
+   specified distance. With multiple matches on both left and right sides it is
+   possible for the cluster of points having a single source identifier to
+   expand in size beyond the distance threshold.
```

Comment:
To me, the critical piece of information is still missing, and that is the each entry in either table gets a single "cluster" ID.

In the standard case I can also assign an identifier to the left and right entries, which defines every pair within the specified distance (what you said above). What is different in this algorithm is that a given entry can have only one such identifier. I'm guessing that is what you meant with the word "common", which I find hard to understand here.

It might be better to say that the algorithm finds "clusters", and assigns each entry in both tables a **unique** cluster identifier which is defined by a transitive relation defined by `join_func`.

In general, distance is not a transitive relation. The requirement that the distance between entries in one table is larger than twice the join distance removes the matches that would break the transitivity.